eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,50 @@
+  [string]$CodeOwnerFileLocation = "../../../.github/CODEOWNERS", # The place we fetch CODEOWNERS file
+  [string]$VsoVariable = "" # Option of write code owners into devop variable
 )
-$target = $TargetDirectory.ToLower().Trim("/")
-$codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
-$ownedFolders = @{}
 
-if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
-  exit 1
+# Check if the retrieve-codeowners tool exsit or not.
+function InstallRetrieveCodeOwnersTool() {
+  if (Get-Command "$ToolPath/retrieve-code-owners") {
+    return
+  }
+  Write-Warning "Installing the retrieve-code-owner tool under $ToolPath... "
+  dotnet tool install --tool-path $toolPath --add-source "$DevOpsFeed" --version "$ToolVersion" "Azure.Sdk.Tools.RetrieveCodeOwners"
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+# Get the json property from tool command.
+function GetCodeOwnersEntryFromCommand() {
+  if (Get-Command "$ToolPath/retrieve-code-owners") {
+    Write-Error "The retieve-code-owners tool is not properly installed. Please check your tool path. $ToolPath"
   }
+  return & "$ToolPath/retrieve-code-owners" --target-directory "$CodeOwnerPathExpression" --code-owner-file-path "$CodeOwnerFileLocation" 
 }
 
-$results = $ownedFolders[$target]
-
-if ($results) {
-  Write-Host "Found a folder $results to match $target"
+function GetCodeOwners([string]$codeOwnersString) 
+{ 
+  $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
+  if (!$codeOwnersJson) {
+    Write-Host "No code owners returned from the path: $CodeOwnerPathExpression"
+    return ""
+  }
   
+  $codeOwners = $codeOwnersJson.Owners -join ","
+  Write-Host "Code owners are $codeOwners"
   if ($VsoVariable) {
-    $alreadyPresent = [System.Environment]::GetEnvironmentVariable($VsoVariable)
-
-    if ($alreadyPresent) { 
-      $results += ",$alreadyPresent"
-    }
-    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$results"
+    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeOwners"
   }
-
-  return $results
+  return $codeOwners
 }
-else {
-  Write-Host "Unable to match path $target in CODEOWNERS file located at $codeOwnersLocation."
-  Write-Host ($ownedFolders | ConvertTo-Json)
+
+InstallRetrieveCodeOwnersTool
+
+$codeOwnerToolOutput = GetCodeOwnersEntryFromCommand
```

Comment:
Why not do this in GetCodeOwners function?

Reply:
Given you aren't really calling these functions more than once you can probably eliminate the functions and just have the code inline.

Reply:
I agree with moving this code inside `GetCodeOwners`. Re: eliminating functions, I disagree. I think a lot of our scripts have grown larger over time, and the lack of functions at the start makes our scripts tend to evolve into spaghetti code (`New-TestResources.ps1` as the best example of this). I think if we try to keep all logic in functions except for a single entrypoint function call at the end of the script, it's easier to modify the code AND easier to test because you can dot source individual functions and execute them locally (provided you filter on [invocation name](https://github.com/Azure/azure-sdk-tools/blob/c71702684afb58dfb847d6a1e1cc3114f48be48a/eng/common/scripts/stress-testing/find-all-stress-packages.ps1#L59)).

Reply:
Yes testing is definitely a great reason to have these functions.